### PR TITLE
Plugin Management UX Improvements

### PR DESCRIPTION
**Pull request by LukeTowers**
With the assistance of @Teranode (#3374), here are the improvements to the Plugin Management UX as required by #2714 & #2774. 

Old Plugin Management Listing Page:
screen shot 2018-02-01 at 8 40 35 pm

New Plugin Management Listing Page:
screen shot 2018-02-01 at 8 41 04 pm

Old Plugin Bulk Management Actions:
screen shot 2018-02-01 at 8 40 43 pm

New Plugin Bulk Management Actions:
screen shot 2018-02-01 at 8 41 19 pm

### CODE REVIEW

**Comment by munxar** [comment]
Fantastic job! I really like it.
One thing I would interpret differently are the icons in the dropdown.
I would switch the disable plugin (pause icon) with the disable updates (disabled icon) *and*
the enable plugin (play icon) with the enable updates (check icon).
Why?
The update process is like something that runs forward as time goes by. By stopping it we actually pausing it, if enabled it will run the updates again (at least on a meta level).
On the other hand the check and disabled icon would more naturally fit to my understanding of enabled, disabled.

For the reset plugins: The bomb indicates something blows up, which is (should) not be the case. As I understand we just force to download the plugins source code again? No migration, so no data loss. Imho it should by a sync icon or something.

Again just my first impression and totally biased, do with this input whatever you like.
Again, good job!

**Comment by LukeTowers** [comment]
@munxar good point on the icons, something to consider. In regards to the bomb icon, it does destroy your data because it's effectively running `down` and then `up` on the plugin's migrations, which is the same as removing the tables completely and recreating them, destroying any data added by the user.

**Comment by LukeTowers** [comment]
Updated actions:

screen shot 2018-02-02 at 12 23 33 am

**Comment by munxar** [comment]
I'd vote only for superusers. Good point!

**Comment by LukeTowers** [comment]
Merged this into develop; I changed it so that only superusers can reset plugin data; and only when the site is in debug mode.

**Comment by alvaro-canepa** [comment]
It would be great to have a button to apply new updates on plugins in development stage. The same as logout/login from backend to apply new updates, without to reload entire version branchs.

**Comment by Eoler** [comment]
>It would be great to have a button to apply new updates on plugins in development stage.

The Rainlab.Builder (Versions) already does it for plugin you're working on.

**Comment by alvaro-canepa** [comment]
@Eoler but you need to install it only for one feature. Not all developers use it. I prefer to build plugins using command line scaffolding.

**Comment by LukeTowers** [comment]
@alvaro-canepa pull requests always welcome :)

**Comment by SiyuQian** [comment]
@LukeTowers Good Work! I have one question about this UX improvement. Why not have a toggle button on the /backend/system/updates to allows the users disable and enable plugins without entering the mange plugin page(backend/system/updates/manage)

**Comment by LukeTowers** [comment]
@SiyuQian honestly because of the effort that it would require. I would have had to redirect the manage action as well as implement all of that work on the normal page, so I went with just getting the improvements out there to start and then probably at a later date we'll go back and move it to the plugin listing page.

**Comment by Teranode** [comment]
I too initally thought about doing it from the updates page, but I was on a supposedly made time crunch from work. Honestly i like where it is, while I like lazy/efficent end goals, i personally prefer to make people make that one extra step :V (as a developer)

**Comment by GinoPane** [comment]
When is it expected to go live?

**Comment by LukeTowers** [comment]
In a few weeks @GinoPane; basically as soon as @daftspunk releases a new build.

File: modules/system/controllers/updates/_list_manage_toolbar.htm [hunk 1]
```diff
@@ -1,52 +1,94 @@
...
-            data-request-confirm="<?= e(trans('system::lang.plugins.refresh_confirm')) ?>"
-            data-trigger-action="enable"
-            data-trigger=".control-list input[type=checkbox]"
-            data-trigger-condition="checked"
-            data-stripe-load-indicator>
-            <?= e(trans('system::lang.plugins.refresh')) ?>
-        </button>
-        -->
+
+        <ul class="dropdown-menu" data-dropdown-title="<?= e(trans('rainlab.user::lang.users.bulk_actions')) ?>">
```
Comment (adrenth) [inline modules/system/controllers/updates/_list_manage_toolbar.htm]: This translation message identifier is invalid.

Comment (LukeTowers) [reply modules/system/controllers/updates/_list_manage_toolbar.htm]: @adrenth could you make a PR to fix that?

**Pull request merged**